src/pages/socialGroups/SocialGroup.jsx

```diff
@@ -0,0 +1,233 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { get } from 'lodash-es';
+
+import Grid from '@material-ui/core/Grid';
+import Paper from '@material-ui/core/Paper';
+import AddIcon from '@material-ui/icons/Add';
+
+import useDocumentTitle from '../../hooks/useDocumentTitle';
+import useSocialGroup from '../../models/socialGroups/useSocialGroup';
+import useQueryIndividualsByGuid from '../../models/individual/useQueryIndividualsByGuid';
+import usePatchSocialGroup from '../../models/socialGroups/usePatchSocialGroup';
+import { formatDate } from '../../utils/formatters';
+import RemoveFromSocialGroupDialog from '../../components/dialogs/RemoveFromSocialGroupDialog';
+import ActionIcon from '../../components/ActionIcon';
+import IndividualsDisplay from '../../components/dataDisplays/IndividualsDisplay';
+import InputRow from '../../components/fields/edit/InputRow';
+import TextInput from '../../components/inputs/TextInput';
+import ErrorDialog from '../../components/dialogs/ErrorDialog';
+import ButtonLink from '../../components/ButtonLink';
+import Button from '../../components/Button';
+import LoadingScreen from '../../components/LoadingScreen';
+import MainColumn from '../../components/MainColumn';
+import SadScreen from '../../components/SadScreen';
+import Text from '../../components/Text';
+import AddMembersDialog from './AddMembersDialog';
+
+const nameSchema = { labelId: 'NAME_OF_GROUP' };
+
+export default function SocialGroup() {
+  const { guid } = useParams();
+  const [addDialogOpen, setAddDialogOpen] = useState(false);
+  const [individualToRemove, setIndividualToRemove] = useState(null);
+
+  const openAddDialog = useCallback(() => setAddDialogOpen(true), []);
+  const handleCloseAddDialog = useCallback(
+    () => setAddDialogOpen(false),
+    [],
+  );
+  const handleCloseDeleteDialog = useCallback(
+    () => setIndividualToRemove(null),
+    [],
+  );
+
+  const { data, loading, error, statusCode } = useSocialGroup(guid);
+  const safeMembers = get(data, 'members', {});
+  const memberGuids = Object.keys(safeMembers);
+  const { data: membersWithData, loading: membersLoading } =
+    useQueryIndividualsByGuid(memberGuids);
+
+  const {
+    mutate: patchSocialGroup,
+    loading: patchLoading,
+    error: patchError,
+    clearError: clearPatchError,
+  } = usePatchSocialGroup();
+
+  const [name, setName] = useState('');
+
+  useEffect(() => {
+    if (data?.name) setName(data.name);
+  }, [data?.name]);
```

Comment:
Instead of forcing a reset of `name` every time `data.name` changes, it might be better to disable the text input while the patch is loading.

Reply:
Good call - I added code to disable the input while the patch is loading. but you still need this useEffect to make sure that it stays consistent (for example if the request goofs and resets the name to `''` or something, you need the input to update to reflect the reality of the situation